### PR TITLE
FreeBSD: fix OSS support

### PR DESCRIPTION
**Pull request by rozhuk-im**
## Description
Fix broken OSS support on FreeBSD, see #16780

## Motivation and Context
OSS is best sound backend on FreeBSD, it is very nice if it will work again.

## How Has This Been Tested?
Build and run on FreeBSD, sound cards detected properly.

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: cmake/scripts/freebsd/ArchSetup.cmake [hunk 1]
```diff
@@ -1,6 +1,6 @@
 set(ARCH_DEFINES -DTARGET_POSIX -DTARGET_FREEBSD)
 set(SYSTEM_DEFINES -D__STDC_CONSTANT_MACROS -D_LARGEFILE64_SOURCE
-                   -D_FILE_OFFSET_BITS=64)
+                   -D_FILE_OFFSET_BITS=64 -DHAS_OSS)
```
Comment (notspiff) [inline cmake/scripts/freebsd/ArchSetup.cmake]: is this not set by buildsystem probe?

Comment (wsnipex) [reply cmake/scripts/freebsd/ArchSetup.cmake]: no, since there is no FindOSS module (anymore), this should be added

**Comment by lrusak** [comment]
Can you move the oss registration into a Freebsd specific file please?

**Comment by t6** [comment]
Is something else needed here?

**Comment by wsnipex** [comment]
Is OSS guaranteed to be available on freebsd? If not, you need to add a cmake find module that checks for it

**Comment by t6** [comment]
> Is OSS guaranteed to be available on freebsd?

Yes, it's part of the base system.

**Comment by t6** [comment]
Ping?

**Comment by lrusak** [comment]
jenkins build this please

**Comment by t6** [comment]
Ping again? The Jenkins build was fine. Anything else missing here or can it finally be merged?

**Comment by fritsch** [comment]
@notspiff @wsnipex Are your comments adressed properly? If yes - please merge it.

**Pull request merged**